.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDurationExtractor.cs

```diff
@@ -220,6 +220,22 @@ private List<ExtractResult> MergeMultipleDuration(string text, List<ExtractResul
                 return extractorResults;
             }
 
+            // If the first and last extractions are both preceded/followed by modifiers,
+            // they should not be merged, e.g. "last 2 weeks and 3 days ago"
+            var minStart = extractorResults.Min(er => er.Start);

```

Comment:
What happens if there are more than two extracted entities here? Can this happen?

Reply:
yes it can happens, there are test cases where more than 2 durations are merged. Now, if the first and last are preceded by modifiers, all the extractions are returned separately. 
I considered merging the intermediate extractions, but it is hard to decide which extractions should be merged. 
For example: "They worked on the problem for the last 3 months and 2 weeks and 2 days ago they found a solution",
it is not clear if "2 weeks" should be merged with "last 3 months" or with "2 days ago". 
But I will try to modify the code to allow merging when the units are not contiguous, e.g. "For the last 3 months, they worked on the problem and 2 weeks and 2 days ago they found a solution", in this case "2 weeks and 2 days ago" can be merged unambiguously.

Reply:
Thanks. Please add a spec example of the 3 entities not merged and one for the non-contiguous case.

Reply:
modified and test cases added